### PR TITLE
Fix validate interface for firewall rule creation to support interface groups

### PR DESCRIPTION
**Pull request by dihedral**
See #241

### CODE REVIEW

**Review by jaredhendrickson13 — CHANGES_REQUESTED** [review]
I've left some context on #241. Essentially this pull request will be considered blocked until an interface group endpoint is developed. You are welcome to leave this pull request open until then so it can be merged in at the same time. There are a few things I'd like to see addressed on this pull request first however.

File: pfSense-pkg-API/files/etc/inc/api/models/APIFirewallRuleCreate.inc
Comment (dihedral) [reply pfSense-pkg-API/files/etc/inc/api/models/APIFirewallRuleCreate.inc]: I understand and agree! The support for firewall rules on interface groups is now moved to the `APITools\get_pfsense_if_id`  function and can be used by an optional flag.

I have also created the Endpoint `APIInterfaceGroup` it supports all the request methods GET, POST, PUT and DELETE.
While doing so I spotted that pfSense supports deleting an interface group while firewall rules are still present. In this case, the rules are still viewable via the API, so they must be lurking around somewhere but not visible through the pfSense ui itself.
I have decided to not included any logic to avoid deleting interface groups with active firewall rules, as pfSense itself supports this behaviour.

**Review by jaredhendrickson13 — APPROVED** [review]

**Comment by jaredhendrickson13** [comment]
This looks great! Thanks for your work. I'll merge these changes into my v1.5.0 branch (#243) so they can be tested along side alongside other features being staged for that release. I will include development builds in that PR periodically that can be used to test until the release is ready.

**Pull request merged**
